File: legacy-python/envx/cli/commands/env.py
"""Environment lifecycle commands."""

from __future__ import annotations

import typer

from envx.cli.commands.common import build_service, handle_error, render_rows
from envx.core.exceptions import EnvxError


def register(app: typer.Typer) -> None:
    """Register environment lifecycle commands."""
    env_app = typer.Typer(help="Manage named environments.")

    @env_app.command("create", help="Create a new environment.")
    def create(name: str = typer.Argument(..., help="Environment name.")) -> None:
        service = build_service()
        try:
            service.create_environment(name)
        except EnvxError as exc:
            handle_error(exc)
        typer.echo(f"Created environment '{name}'.")

    @env_app.command("use", help="Set the active environment.")
    def use(name: str = typer.Argument(..., help="Environment name.")) -> None:
        service = build_service()
        try:
            service.use_environment(name)
        except EnvxError as exc:
            handle_error(exc)
        typer.echo(f"Active environment is now '{name}'.")

    @env_app.command("delete", help="Delete an environment.")
    def delete(name: str = typer.Argument(..., help="Environment name.")) -> None:
        service = build_service()
        try:
            service.delete_environment(name)
        except EnvxError as exc:
            handle_error(exc)
        typer.echo(f"Deleted environment '{name}'.")

    @env_app.command("list", help="List available environments.")
    def list_envs() -> None:
        service = build_service()
        try:
            rows = service.list_environments()
        except EnvxError as exc:
            handle_error(exc)
        render_rows(rows, "table")

    app.add_typer(env_app, name="env")

File: internal/cli/env.go
package cli

import (
	"errors"
	"fmt"
	"io"
)

func cmdEnv(args []string, stdout, stderr io.Writer) int {
	root, rest := splitRootFlag(args)
	if checkHelp(rest, stdout, "env") {
		return 0
	}
	if len(rest) == 0 {
		return printError(stderr, errors.New("usage: envx env create|use|delete|list [NAME]"))
	}
	sub := rest[0]
	rest = rest[1:]
	service := projectService(root)

	switch sub {
	case "create":
		if len(rest) != 1 {
			return printError(stderr, errors.New("usage: envx env create NAME"))
		}
		if err := service.CreateEnvironment(rest[0]); err != nil {
			return printError(stderr, err)
		}
		fmt.Fprintf(stdout, "Created environment '%s'.\n", rest[0])
		return 0
	case "use":
		if len(rest) != 1 {
			return printError(stderr, errors.New("usage: envx env use NAME"))
		}
		if err := service.UseEnvironment(rest[0]); err != nil {
			return printError(stderr, err)
		}
		fmt.Fprintf(stdout, "Active environment is now '%s'.\n", rest[0])
		return 0
	case "delete":
		if len(rest) != 1 {
			return printError(stderr, errors.New("usage: envx env delete NAME"))
		}
		if err := service.DeleteEnvironment(rest[0]); err != nil {
			return printError(stderr, err)
		}
		fmt.Fprintf(stdout, "Deleted environment '%s'.\n", rest[0])
		return 0
	case "list":
		rows, err := service.ListEnvironments()
		if err != nil {
			return printError(stderr, err)
		}
		renderEnvironmentRows(rows, stdout)
		return 0
	default:
		return printError(stderr, fmt.Errorf("unknown env subcommand '%s'", sub))
	}
}
